spring-kafka/src/main/java/org/springframework/kafka/config/AbstractKafkaListenerContainerFactory.java

```diff
@@ -379,13 +380,27 @@ else if (this.autoStartup != null) {
 						instance.getContainerProperties()::setConsumerProperties);
 	}
 
+	/**
+	 * {@inheritDoc}
+	 * @deprecated in favor of {@link #createContainer(TopicPartitionOffset[])}
+	 */
+	@Deprecated
+	@SuppressWarnings("deprecation")
+	@Override
+	public C createContainer(final Collection<org.springframework.kafka.support.TopicPartitionInitialOffset> topicPartitions) {
+		return createContainer(topicPartitions.stream()
+				.map(TopicPartitionOffset::fromTPIO)
+				.collect(Collectors.toList())
+				.toArray(new TopicPartitionOffset[0]));
+	}
+
 	@Override
-	public C createContainer(final Collection<TopicPartitionOffset> topicPartitions) {
+	public C createContainer(final TopicPartitionOffset[] topicsAndPartitions) {
```

Comment:
Can this be a `varargs`?

Reply:
No; `@SafeVarArgs` can't be used on a non-final method.

Reply:
My plan is to deprecate this in 2.4 and reintroduce `Collection<TPO>`.

Reply:
Why do we talk about `@SafeVarArgs` here if our type is just an explicit and there is no any generics?
😕

Reply:
Not sure what happened, but I got a warning yesterday; now it's not complaining - fixing.